Fix - API Create Relationship via Link

Jason-Dang opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here unless your commit contains the issue number -->

- Add link option to create relationship params and route.
- Use given link instead of guessing from beans.
- Update Postman collection with new path part to specify link name.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

Previously unable to specify link and logic just grabbed the last link it found between two modules.

## How To Test This
<!--- Please describe in detail how to test your changes. -->

Use the updated Create Relationship call in the Postman collection.
Example: For `Accounts` use the `contacts` link.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
<!--- Go over all the following points and check all the boxes that apply. --->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! --->
- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [x] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

<!--- Your pull request will be tested via Travis CI to automatically indicate that your changes do not prevent compilation. --->

<!--- If it reports back that there are problems, you can log into the Travis system and check the log report for your pull request to see what the problem was. --->

PedroErnst commented on Api/V8/Config/routes.php:
```diff
@@ -98,7 +98,7 @@
         $app
             ->post(
-                '/module/{moduleName}/{id}/relationships',
+                '/module/{moduleName}/{id}/relationships/{linkFieldName}',
```
Would it maybe make sense to leave the old route active as well?  
I think by changing the route it becomes a breaking change that's not backward compatible. Existing applications that use this API might stop working.  
It might be good to have both routes.

Jason-Dang replied:
That’s a good point, although personally I would argue it not to be a breaking change as the current route never worked as it should have. I don’t know if anyone would want an endpoint that creates a random link between two modules if they have more than one relationship. Well I believe it just uses the first/last link it finds. 

I do see the sense in maintaining backward compatibility and will be happy with whatever decision is made.

Dillon-Brown commented:
@Jason-Dang Failing tests/api/V8/CreateRelationShipCest.

Jason-Dang commented:
@Dillon-Brown I forgot about the test, I believe it should be updated to use an out of the box link name such as `contacts` for the relationship between Accounts and Contacts.

I'll push up a commit with this update.

Yeah that is strange as IMO it's how it should be.

Jason-Dang commented:
@Dillon-Brown Finally made the time to fix this properly. 
I fixed the test and it seemed to pass travis so I've now added some extra meta info and changed it to use module labels. 
Currently with travis now so hopefully all good and this can be merged.

pgorod commented:
@Jason-Dang or @PedroErnst, if any of you guys can please have a quick look at this Documentation PR, it's also about using relationships via the API so it's similar to what you're doing here and you should be able to review that in just a minute.

https://github.com/salesagility/SuiteDocs/pull/318/files

Please just leave a positive review there if it looks ok. Thanks!

Jason-Dang commented:
@Dillon-Brown I've reverted the old endpoint and added a new one with the changes. It's going through travis now. I'll leave the deprecation of the old one up to your team. I've also updated the Postman collection to remove all the dummy access token values and replaced ID values with the placeholder `{ID}`.

Jason-Dang commented:
@pgorod I'll take a look soon as I get some free time, would you mind nudging me with a reminder in case I forget down the line please.

Dillon-Brown commented:
Assessed
